### PR TITLE
Release 3.1.0

### PR DESCRIPTION
**Pull request by jordyn**
Release 3.1.0

### CODE REVIEW

**Review by bmc-msft — APPROVED** [review]

**Review by bmc-msft — COMMENTED** [review]
Note, I tried to order the sections (Fixed/Added/Changed) alphabetically to be consistent across releases.

**Review by chkeita — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,22 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 3.1.0
+### Fixed
+* Deployment: Fixed an issue with multitenant deployment caused by a mismatch between the identifier used to configure the app registration and value used to authenticate the CLI client. [#1270](https://github.com/microsoft/onefuzz/pull/1270)
```
Comment (mgreisen) [inline CHANGELOG.md:9]: multitenant ==> multi-tenant

**Review by mgreisen — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,22 @@ All notable changes to this project will be documented in this file.
...
+* Service: Fixed `scaleset proxy reset` to reset all proxies in specified region. [#1275](https://github.com/microsoft/onefuzz/pull/1275)
+* CLI: Temporarily ignoring type errors from `azure-storage-blob` due to invalid python type signatures. [#1258](https://github.com/microsoft/onefuzz/pull/1258)
```
Comment (ranweiler) [inline CHANGELOG.md:11]:
```suggestion
* CLI: Temporarily ignore type errors from `azure-storage-blob` due to invalid python type signatures. [#1258](https://github.com/microsoft/onefuzz/pull/1258)
```

File: CHANGELOG.md (hunk 1 above)
Comment (ranweiler) [reply CHANGELOG.md:9]:
```suggestion
* Deployment: Fixed an issue with multi-tenant deployment caused by a mismatch between the identifier used to configure the app registration and value used to authenticate the CLI client. [#1270](https://github.com/microsoft/onefuzz/pull/1270)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,22 @@ All notable changes to this project will be documented in this file.
...
+
+### Added
+* Service: Added ability to configure virtual network IP ranges. [#1268](https://github.com/microsoft/onefuzz/pull/1268)
+* Deployment: Added `flake8` to the deployment process to align with rest of the python codebase linting process. [#1286](https://github.com/microsoft/onefuzz/pull/1286)
```
Comment (ranweiler) [inline CHANGELOG.md:15]:
```suggestion
* Deployment: Added `flake8` to the deployment process to align with rest of the Python codebase linting. [#1286](https://github.com/microsoft/onefuzz/pull/1286)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,22 @@ All notable changes to this project will be documented in this file.
...
+* Service: Added custom extensions to enable Microsoft Security Monitoring extensions. [#1184](https://github.com/microsoft/onefuzz/pull/1184)
+* CLI: Added `--readonly_inputs` option for `onefuzz templates libfuzzer` command. [#1247](https://github.com/microsoft/onefuzz/pull/1247)
```
Comment (ranweiler) [inline CHANGELOG.md:17]:
```suggestion
* CLI: Added `--readonly_inputs` option to the `libfuzzer_basic` template. [#1247](https://github.com/microsoft/onefuzz/pull/1247)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,22 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 3.1.0
+### Fixed
```
Comment (ranweiler) [inline CHANGELOG.md:8]: As @bmc-msft suggested, it'd be nice to preserve consistent section ordering (when many are present). This is just alphabetical: "Added", "Changed", "Fixed".

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,22 @@ All notable changes to this project will be documented in this file.
...
+* Deployment: Fixed an issue with multi-tenant deployment caused by a mismatch between the identifier used to configure the app registration and value used to authenticate the CLI client. [#1270](https://github.com/microsoft/onefuzz/pull/1270)
+* Service: Fixed `scaleset proxy reset` to reset all proxies in specified region. [#1275](https://github.com/microsoft/onefuzz/pull/1275)
+* CLI: Temporarily ignore type errors from `azure-storage-blob` due to invalid Python type signatures. [#1258](https://github.com/microsoft/onefuzz/pull/1258)
+
+### Added
+* Service: Added ability to configure virtual network IP ranges. [#1268](https://github.com/microsoft/onefuzz/pull/1268)
+* Deployment: Added `flake8` to the deployment process to align with rest of the python codebase linting process. [#1286](https://github.com/microsoft/onefuzz/pull/1286)
+* Service: Added custom extensions to enable Microsoft Security Monitoring extensions. [#1184](https://github.com/microsoft/onefuzz/pull/1184)
+* CLI: Added `--readonly_inputs` option to the `libfuzzer_basic` template. [#1247](https://github.com/microsoft/onefuzz/pull/1247)
```
Comment (ranweiler) [inline CHANGELOG.md:21]: Sorry, my original suggestion had a typo! 😅 
```suggestion
* CLI: Added `--readonly_inputs` option to the `libfuzzer basic` template. [#1247](https://github.com/microsoft/onefuzz/pull/1247)
```

File: CHANGELOG.md (hunk 3 above)
Comment (ranweiler) [reply CHANGELOG.md:19]: (I think this one got lost somehow when you rearranged the sections)

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,23 @@ All notable changes to this project will be documented in this file.
...
 
+## 3.1.0
+### Changed
+* CLI: Increased the default verbosity of destructive CLI commands. [#1264](https://github.com/microsoft/onefuzz/pull/1264)
+* Agent/Supervisor/Proxy: Updated multiple third-party Rust dependencies. [#1239](https://github.com/microsoft/onefuzz/pull/1239), [#1240](https://github.com/microsoft/onefuzz/pull/1240), [#1236](https://github.com/microsoft/onefuzz/pull/1236), [#1238](https://github.com/microsoft/onefuzz/pull/1238), [#1245](https://github.com/microsoft/onefuzz/pull/1245), [#1246](https://github.com/microsoft/onefuzz/pull/1246), [#1252](https://github.com/microsoft/onefuzz/pull/1252), [#1253](https://github.com/microsoft/onefuzz/pull/1253), [#1254](https://github.com/microsoft/onefuzz/pull/1254), [#1257](https://github.com/microsoft/onefuzz/pull/1257), [#1261](https://github.com/microsoft/onefuzz/pull/1261), [#1262](https://github.com/microsoft/onefuzz/pull/1262), [#1276](https://github.com/microsoft/onefuzz/pull/1276), [#1278](https://github.com/microsoft/onefuzz/pull/1278)
+
+### Fixed
+* Deployment: Fixed deployment issues in the `westus3` region due to `Region doesn't support Classic resource mode for Application Insights resources` error.
```
Comment (ranweiler) [inline CHANGELOG.md:13]:
```suggestion
* Deployment: Fixed deployment in some regions by deploying newer, broadly-supported versions of Application Insights resources. [#1291](https://github.com/microsoft/onefuzz/pull/1291)
```

**Review by mgreisen — APPROVED** [review]

**Pull request merged**